// esphome-config-ts/dist/components/copy.js
/**
 * This file was automatically generated.
 * DO NOT MODIFY BY HAND.
 *
 * Schema: copy.json
 * Source: https://github.com/esphome/esphome/blob/dev/esphome/components/copy
 *
 * This file is part of the ESPHome TypeScript Generator.
 * © Allport-IT Ltd (t/a Local Bytes)
 **/
import { EsphomeComponent } from "../lib/base.js";
export class CopySensor extends EsphomeComponent {
    componentName = "copy.sensor";
}
export class CopyBinarySensor extends EsphomeComponent {
    componentName = "copy.binary_sensor";
}
export class CopyButton extends EsphomeComponent {
    componentName = "copy.button";
}
export class CopyCover extends EsphomeComponent {
    componentName = "copy.cover";
}
export class CopyFan extends EsphomeComponent {
    componentName = "copy.fan";
}
export class CopyLock extends EsphomeComponent {
    componentName = "copy.lock";
}
export class CopyNumber extends EsphomeComponent {
    componentName = "copy.number";
}
export class CopySelect extends EsphomeComponent {
    componentName = "copy.select";
}
export class CopySwitch extends EsphomeComponent {
    componentName = "copy.switch";
}
export class CopyTextSensor extends EsphomeComponent {
    componentName = "copy.text_sensor";
}
