// esphome-config-ts/dist/components/custom.js
/**
 * This file was automatically generated.
 * DO NOT MODIFY BY HAND.
 *
 * Schema: custom.json
 * Source: https://github.com/esphome/esphome/blob/dev/esphome/components/custom
 *
 * This file is part of the ESPHome TypeScript Generator.
 * © Allport-IT Ltd (t/a Local Bytes)
 **/
import { EsphomeComponent } from "../lib/base.js";
export class CustomSensor extends EsphomeComponent {
    componentName = "custom.sensor";
}
export class CustomBinarySensor extends EsphomeComponent {
    componentName = "custom.binary_sensor";
}
export class CustomClimate extends EsphomeComponent {
    componentName = "custom.climate";
}
export class CustomCover extends EsphomeComponent {
    componentName = "custom.cover";
}
export class CustomLight extends EsphomeComponent {
    componentName = "custom.light";
}
export class CustomOutput extends EsphomeComponent {
    componentName = "custom.output";
}
export class CustomSwitch extends EsphomeComponent {
    componentName = "custom.switch";
}
export class CustomTextSensor extends EsphomeComponent {
    componentName = "custom.text_sensor";
}
