// esphome-config-ts/dist/components/ld2410.js
/**
 * This file was automatically generated.
 * DO NOT MODIFY BY HAND.
 *
 * Schema: ld2410.json
 * Source: https://github.com/esphome/esphome/blob/dev/esphome/components/ld2410
 *
 * This file is part of the ESPHome TypeScript Generator.
 * © Allport-IT Ltd (t/a Local Bytes)
 **/
import { EsphomeComponent } from "../lib/base.js";
export class Ld2410 extends EsphomeComponent {
    componentName = "ld2410";
}
export class Ld2410Sensor extends EsphomeComponent {
    componentName = "ld2410.sensor";
}
export class Ld2410BinarySensor extends EsphomeComponent {
    componentName = "ld2410.binary_sensor";
}
export class Ld2410Button extends EsphomeComponent {
    componentName = "ld2410.button";
}
export class Ld2410Number extends EsphomeComponent {
    componentName = "ld2410.number";
}
export class Ld2410Select extends EsphomeComponent {
    componentName = "ld2410.select";
}
export class Ld2410Switch extends EsphomeComponent {
    componentName = "ld2410.switch";
}
export class Ld2410TextSensor extends EsphomeComponent {
    componentName = "ld2410.text_sensor";
}
