// esphome-config-ts/dist/components/modbus_controller.js
/**
 * This file was automatically generated.
 * DO NOT MODIFY BY HAND.
 *
 * Schema: modbus_controller.json
 * Source: https://github.com/esphome/esphome/blob/dev/esphome/components/modbus_controller
 *
 * This file is part of the ESPHome TypeScript Generator.
 * © Allport-IT Ltd (t/a Local Bytes)
 **/
import { EsphomeComponent } from "../lib/base.js";
export class ModbusController extends EsphomeComponent {
    componentName = "modbus_controller";
}
export class ModbusControllerSensor extends EsphomeComponent {
    componentName = "modbus_controller.sensor";
}
export class ModbusControllerBinarySensor extends EsphomeComponent {
    componentName = "modbus_controller.binary_sensor";
}
export class ModbusControllerNumber extends EsphomeComponent {
    componentName = "modbus_controller.number";
}
export class ModbusControllerOutput extends EsphomeComponent {
    componentName = "modbus_controller.output";
}
export class ModbusControllerSelect extends EsphomeComponent {
    componentName = "modbus_controller.select";
}
export class ModbusControllerSwitch extends EsphomeComponent {
    componentName = "modbus_controller.switch";
}
export class ModbusControllerTextSensor extends EsphomeComponent {
    componentName = "modbus_controller.text_sensor";
}
