// esphome-config-ts/dist/components/template.js
/**
 * This file was automatically generated.
 * DO NOT MODIFY BY HAND.
 *
 * Schema: template.json
 * Source: https://github.com/esphome/esphome/blob/dev/esphome/components/template
 *
 * This file is part of the ESPHome TypeScript Generator.
 * © Allport-IT Ltd (t/a Local Bytes)
 **/
import { EsphomeComponent } from "../lib/base.js";
export class TemplateSensor extends EsphomeComponent {
    componentName = "template.sensor";
}
export class TemplateAlarmControlPanel extends EsphomeComponent {
    componentName = "template.alarm_control_panel";
}
export class TemplateBinarySensor extends EsphomeComponent {
    componentName = "template.binary_sensor";
}
export class TemplateButton extends EsphomeComponent {
    componentName = "template.button";
}
export class TemplateCover extends EsphomeComponent {
    componentName = "template.cover";
}
export class TemplateLock extends EsphomeComponent {
    componentName = "template.lock";
}
export class TemplateNumber extends EsphomeComponent {
    componentName = "template.number";
}
export class TemplateOutput extends EsphomeComponent {
    componentName = "template.output";
}
export class TemplateSelect extends EsphomeComponent {
    componentName = "template.select";
}
export class TemplateSwitch extends EsphomeComponent {
    componentName = "template.switch";
}
export class TemplateTextSensor extends EsphomeComponent {
    componentName = "template.text_sensor";
}
