// esphome-config-ts/dist/components/tuya.js
/**
 * This file was automatically generated.
 * DO NOT MODIFY BY HAND.
 *
 * Schema: tuya.json
 * Source: https://github.com/esphome/esphome/blob/dev/esphome/components/tuya
 *
 * This file is part of the ESPHome TypeScript Generator.
 * © Allport-IT Ltd (t/a Local Bytes)
 **/
import { EsphomeComponent } from "../lib/base.js";
export class Tuya extends EsphomeComponent {
    componentName = "tuya";
}
export class TuyaSensor extends EsphomeComponent {
    componentName = "tuya.sensor";
}
export class TuyaBinarySensor extends EsphomeComponent {
    componentName = "tuya.binary_sensor";
}
export class TuyaClimate extends EsphomeComponent {
    componentName = "tuya.climate";
}
export class TuyaCover extends EsphomeComponent {
    componentName = "tuya.cover";
}
export class TuyaFan extends EsphomeComponent {
    componentName = "tuya.fan";
}
export class TuyaLight extends EsphomeComponent {
    componentName = "tuya.light";
}
export class TuyaNumber extends EsphomeComponent {
    componentName = "tuya.number";
}
export class TuyaSelect extends EsphomeComponent {
    componentName = "tuya.select";
}
export class TuyaSwitch extends EsphomeComponent {
    componentName = "tuya.switch";
}
export class TuyaTextSensor extends EsphomeComponent {
    componentName = "tuya.text_sensor";
}
